core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -153,8 +153,13 @@ class KryoSerializer(conf: SparkConf)
     kryo.register(classOf[SerializableJobConf], new KryoJavaSerializer())
     kryo.register(classOf[PythonBroadcast], new KryoJavaSerializer())
 
-    kryo.register(classOf[GenericRecord], new GenericAvroSerializer(avroSchemas))
-    kryo.register(classOf[GenericData.Record], new GenericAvroSerializer(avroSchemas))
+    def registerAvro[T >: Null <: GenericContainer](implicit ct: ClassTag[T]) =
```

Comment:
The method has side effect, change to:

```suggestion
    def registerAvro[T <: GenericContainer]()(implicit ct: ClassTag[T]): Unit =
      kryo.register(ct.runtimeClass, new GenericAvroSerializer[T](avroSchemas))
```

Reply:
+1 good call!

Reply:
Done